runtime/src/main/java/org/corfudb/protocols/wireprotocol/TokenRequest.java

```diff
@@ -78,6 +81,18 @@ public TokenRequest(Long numTokens, List<UUID> streams) {
         txnResolution = null;
     }
 
+    /**
+     * Constructor for generating an Ordered Globally Unique Id TokenRequest.
+     * {@link TokenRequest::TK_OGUID}
```

Comment:
We can use a write token for uniqueness, please don't add this new API to the sequencer.

Reply:
Also, note that after you acquire the token, you need to write to it, to prevent the sequence regression.

Reply:
I thought the whole idea of the new api was to avoid the write and the hole fills etc?

Reply:
The hole-issue is only a problem when we bump the sequencer per enqueue, but if you do it once every time a client boots, then it wont be a problem, since clients are not expected to restart at a really high rate.

Reply:
Sure, but we are not making an assumption on the usage, right? Also what is the real issue with the new API? Its just another switch case in an existing call right?

Reply:
Its not about code complexity, but since its not really necessary why add a new API?

Reply:
New API avoids a write why add a write when we don't need to? Moreover this was your idea to begin with :)

Reply:
Will remove the api and use the token approach you suggest instead. thanks